pkg/system/phase4_configuring.go

```diff
@@ -537,16 +541,156 @@ func (r *Reconciler) awaitEndpointDeploymentPods() error {
 func (r *Reconciler) ReconcileHPAEndpoint() error {
 	// Wait for the the endpoint deployment to become ready
 	// only if HPA was not created yet
+	log := r.Logger.WithField("func", "ReconcileHPAEndpoint")
 	if r.HPAEndpoint.UID == "" {
 		if err := r.awaitEndpointDeploymentPods(); err != nil {
 			return err
 		}
 	}
+	var err error
+	prometheusList := deployedPrometheus(log)
+	switch r.NooBaa.Spec.AutoscalingType {
+	case "keda":
+		if err = r.deleteExistingAutoscaler(string(r.NooBaa.Spec.AutoscalingType)); err != nil {
```

Comment:
`deleteExistingAutoscaler` is relevant for all cases, so maybe call it once before the switch.
Also, the name is a bit confusing. Maybe rename it to something like `cleanPreviousAutoscalingResources`? Something that better state that this is a cleanup code